lib/components/tabs/tabs.dart

```diff
@@ -0,0 +1,136 @@
+import 'dart:html';
+import 'dart:async';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('Tabs');
+
+class TabView {
+  String id;
+  String label;
+  DivElement content;
+
+  TabView(this.id, this.label, this.content);
+}
+
+class TabsView {
+  DivElement renderElement;
+  List<TabView> _tabs;
+  Map<String, TabView> _tabsLookup;
```

Comment:
Since we need to compute tabs lookup whenever we need to use it I suggest we turn this into a getter:
```suggestion
  Map<String, TabView> get _tabsById => Map.fromEntries(_tabs.map((t) => MapEntry(t.id, t)));
```
We can now remove the `_updateLookup` method below, and just call this `_tabsById` whenever we need to, without having to remember to update it beforehand.